plugins/chat/app/models/chat/channel_custom_field.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Chat
+  class ChannelCustomField < ActiveRecord::Base
+    belongs_to :channel
+  end
+end
+
+# == Schema Information
+#
+# Table name: chat_channel_custom_fields
+#
+#  id              :bigint           not null, primary key
+#  chat_channel_id :bigint           not null
+#  name            :string(256)      not null
```

Comment:
I think we should add a unique index on these columns, so that we avoid the kind of race conditions which plague us in other custom_fields.

(core is moving this way - array custom fields were deprecated last year)